datastream/datastream.py

```diff
@@ -213,6 +216,16 @@ def sample_proportion(
             self.sampler.sample_proportion(proportion),
         )
 
+    def take(
+        self: Datastream[T],
+        n_samples: int,
+    ) ->  Datastream[T]:
+        '''
+        Like :func:`Datastream.sample_proportion` but specify the number of
+        samples instead of a proportion.
+        '''
+        return self.sample_proportion(min(1, n_samples / len(self)))
```

Comment:
We allow empty datastreams right? (even if its not super useful)
I know its a degenerate case, but this crashes kinda gnarly if `len(self) == 0`
Also I think `len(self) == 0` and `n_samples == 0` should return an empty datastream

Reply:
We do not support empty datastreams. The most common case where this fails for me has been when merging empty datastreams

Reply:
Added check and tests